projects/gloo/pkg/plugins/als/plugin.go

```diff
@@ -63,33 +67,136 @@ func (p *plugin) ProcessHcmNetworkFilter(params plugins.Params, parentListener *
 func ProcessAccessLogPlugins(service *als.AccessLoggingService, logCfg []*envoyal.AccessLog) ([]*envoyal.AccessLog, error) {
 	results := make([]*envoyal.AccessLog, 0, len(service.GetAccessLog()))
 	for _, al := range service.GetAccessLog() {
+
+		var newAlsCfg envoyal.AccessLog
+
+		// Make the "base" config with output destination
 		switch cfgType := al.GetOutputDestination().(type) {
 		case *als.AccessLog_FileSink:
 			var cfg envoyalfile.FileAccessLog
-			if err := copyFileSettings(&cfg, cfgType); err != nil {
+			err := copyFileSettings(&cfg, cfgType)
+			if err != nil {
 				return nil, err
 			}
-			newAlsCfg, err := translatorutil.NewAccessLogWithConfig(wellknown.FileAccessLog, &cfg)
+
+			newAlsCfg, err = translatorutil.NewAccessLogWithConfig(wellknown.FileAccessLog, &cfg)
 			if err != nil {
 				return nil, err
 			}
-			results = append(results, &newAlsCfg)
+
 		case *als.AccessLog_GrpcService:
 			var cfg envoygrpc.HttpGrpcAccessLogConfig
-			if err := copyGrpcSettings(&cfg, cfgType); err != nil {
+			err := copyGrpcSettings(&cfg, cfgType)
+			if err != nil {
 				return nil, err
 			}
-			newAlsCfg, err := translatorutil.NewAccessLogWithConfig(wellknown.HTTPGRPCAccessLog, &cfg)
+
+			newAlsCfg, err = translatorutil.NewAccessLogWithConfig(wellknown.HTTPGRPCAccessLog, &cfg)
 			if err != nil {
 				return nil, err
 			}
-			results = append(results, &newAlsCfg)
 		}
+
+		// Create and add the filter
+		filter := al.GetFilter()
+		envoyFilter, err := translateFilter(filter)
+		if err != nil {
+			return nil, err
+		}
+		newAlsCfg.Filter = envoyFilter
+		results = append(results, &newAlsCfg)
+
 	}
+
 	logCfg = append(logCfg, results...)
 	return logCfg, nil
 }
 
+// Since we are using the same proto def, marshal out of gloo format and unmarshal into envoy format
+func translateFilter(inFilter *als.AccessLogFilter) (*envoyal.AccessLogFilter, error) {
+	if inFilter == nil {
+		return nil, nil
+	}
+
+	// We need to validate the enums in the filter manually because the protobuf libraries
+	// do not validate them, for "compatibilty reasons". It's nicer to catch them here instead
+	// of sending bad configs to Envoy.
+	if err := validateFilterEnums(inFilter); err != nil {
+		return nil, err
+	}
+
+	bytes, err := proto.Marshal(inFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	outFilter := &envoyal.AccessLogFilter{}
+	if err := proto.Unmarshal(bytes, outFilter); err != nil {
+		return nil, err
+	}
+
+	return outFilter, nil
+}
+
+var (
+	InvalidEnumValueError = func(filterName string, fieldName string, value string) error {
+		return errors.Errorf("Invalid value of %s in Enum field %s of %s", value, fieldName, filterName)
+	}
+	NestedError = func(filterName string, err error) error {
+		return errors.Errorf("%s, inside an %s", err, filterName)
+	}
```

Comment:
By doing this, we can check for error via `errors.Is(err, InvalidEnumValueError)` or in tests with `Expect(err).Should(MatchError(importname.InvalidEnumValueError))` instead of relying on raw string comparison
```suggestion
var (
        InvalidEnumValueError = errors.New("invalid value in Enum Field")
	WrapInvalidEnumValueError = func(filterName string, fieldName string, value string) error {
		return errors.Wrap(InvalidEnumValueError, "invalid value of %s in Enum field %s of %s", value, fieldName, filterName)
	}
```